src/Symfony/Component/Messenger/MessageRecorder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Component\Cache\ResettableInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+class MessageRecorder implements MessageRecorderInterface, RecordedMessageContainerInterface, ResettableInterface
```

Comment:
What do you think about creating a trait like `SimpleBus`. This is useful, especially for domain model.

Reply:
I agree. We should also create a doctrine event subscriber that implements `RecordedMessageContainerInterface`. But that should be a separate PR. Do you want to create that those two?

Reply:
Yes we can create create it in another PR. I can do it! I don't get you about creating a doctrine subscriber.

Reply:
Awesome. Please ping me in that PR and I'll be happy to review. 

About doctrine subscriber. We need (just like simplebus) something that takes messages from your domain objects and add them to a message bus.